src/Microsoft.Identity.Web/AppServicesAuth/AppServicesAuthenticationTokenAcquisition.cs

```diff
@@ -87,16 +106,29 @@ private async Task<IConfidentialClientApplication> GetOrCreateApplication()
         }
 
         /// <inheritdoc/>
-        public async Task<string> GetAccessTokenForUserAsync(
+        public Task<string> GetAccessTokenForUserAsync(
             IEnumerable<string> scopes,
             string? tenantId = null,
             string? userFlow = null,
             ClaimsPrincipal? user = null,
             TokenAcquisitionOptions? tokenAcquisitionOptions = null)
         {
-            string accessToken = GetAccessToken(CurrentHttpContext?.Request.Headers);
+            var httpContext = CurrentHttpContext;
```

Comment:
nit: do you need this extra variable? Can't you just lock on the CurrentHttpContext?

Reply:
It is very unlikely, but still possible that subsequent calls to CurrentHttpContext would yield a different instance. This can happen if someone registers a custom IHttpContextAccessor. For that reason, the httpContext is stored in a var and reused.